File: app/src/main/java/com/pleon/buyt/model/Purchase.java
package com.pleon.buyt.model;

import java.util.Date;

import androidx.room.Entity;
import androidx.room.ForeignKey;
import androidx.room.PrimaryKey;

@Entity
public class Purchase {

    @PrimaryKey(autoGenerate = true)
    private long purchaseId;

    @ForeignKey(entity = Store.class, parentColumns = "id", childColumns = "storeId")
    private long storeId;

    private Date date;

    public Purchase(long storeId, Date date) {
        this.storeId = storeId;
        this.date = date;
    }

    public long getPurchaseId() {
        return purchaseId;
    }

    public void setPurchaseId(long purchaseId) {
        this.purchaseId = purchaseId;
    }

    public long getStoreId() {
        return storeId;
    }

    public void setStoreId(long storeId) {
        this.storeId = storeId;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }
}

File: app/src/main/kotlin/com/pleon/buyt/model/Purchase.kt
package com.pleon.buyt.model

import androidx.room.Entity
import androidx.room.PrimaryKey
import java.util.*

@Entity(
    /* TODO: Uncomment this and migrate the database */
    // foreignKeys = [
    //     ForeignKey(entity = Store::class, parentColumns = ["storeId"], childColumns = ["storeId"])
    // ]
)
class Purchase(val date: Date) {

    @PrimaryKey(autoGenerate = true)
    var purchaseId: Long = 0
    var storeId: Long = 0
}
